### DISCUSSION TITLE
Getting to work backgroundDrawable in JAVA

### DISCUSSION THREAD

**Discussion by ankandrew** [AppIntro/AppIntro#1016] [Q&A] [answered]
### Problem

I want to use a gradient background in JAVA, but the `backgroundColorRes` has priority over the `backgroundDrawable`.

### Sample to Reproduce

I've used the [Example App](https://github.com/AppIntro/AppIntro/tree/main/example) and changed `JavaIntro` first lines:

#### From

https://github.com/AppIntro/AppIntro/blob/0b782dbcfe0c4f0fa43b0b87bc736852c5e1ab40/example/src/main/java/com/github/appintro/example/ui/java/JavaIntro.java#L19-L21

#### To

```java
addSlide(AppIntroFragment.createInstance(
        "Welcome!",
        "This is a demo example in java of AppIntro library, with a custom background on each slide!",
        R.drawable.ic_slide1,
        0,
        0,
        0,
        0,
        0,
        R.drawable.back_slide1
));
```

### Result

![photo_2022-07-06_19-31-46](https://user-images.githubusercontent.com/61120139/177654269-e0686722-586a-4228-b946-493979915243.jpg)

**Comment by ankandrew** [comment]
@cortinico Am I doing something wrong?

**Comment by cortinico** [comment, marked as answer]
What you're doing is correct. However, `backgroundDrawable` has priority over the other background:

https://github.com/AppIntro/AppIntro/blob/3a301e4021e3a81c8684defc63f06bb623afc3b5/appintro/src/main/java/com/github/appintro/AppIntroBaseFragment.kt#L155-L166

If you wish me to investigate further, you need to provide more details (like a reproducer, which versions of AppIntro you're using, and the content of `back_slide1`).

**Reply by ankandrew** [reply]
### Sample to Reproduce

I've used the [Example App](https://github.com/AppIntro/AppIntro/tree/main/example) and changed `JavaIntro` first lines:

#### From

https://github.com/AppIntro/AppIntro/blob/0b782dbcfe0c4f0fa43b0b87bc736852c5e1ab40/example/src/main/java/com/github/appintro/example/ui/java/JavaIntro.java#L19-L21

#### To

```java
addSlide(AppIntroFragment.createInstance(
        "Welcome!",
        "This is a demo example in java of AppIntro library, with a custom background on each slide!",
        R.drawable.ic_slide1,
        0,
        0,
        0,
        0,
        0,
        R.drawable.back_slide1
));
```

So AppIntro version must be the lastest one and `back_slide1.xml` can be found [in this repo](https://github.com/AppIntro/AppIntro/blob/main/example/src/main/res/drawable/back_slide1.xml) (didn't change).

**Reply by ankandrew** [reply]
I can create a repo to reproduce, but that small change is all I have changed to get the results

**Reply by cortinico** [reply]
What is happening is that Color Transition is enabled for the Java Example:
https://github.com/AppIntro/AppIntro/blob/0b782dbcfe0c4f0fa43b0b87bc736852c5e1ab40/example/src/main/java/com/github/appintro/example/ui/java/JavaIntro.java#L50-L51

This overrides the drawable background you specified as the AppIntro is effectively running a color transition between two black slides.

If you comment the line I mentioned, it will work fine 👍

**Reply by ankandrew** [reply]
Yep, that works. Thank you
